pkg/api/frontend_logging.go

```diff
@@ -54,40 +53,63 @@ func GrafanaJavascriptAgentLogMessageHandler(store *frontendlogging.SourceMapSto
 		if err := web.Bind(c.Req, &event); err != nil {
 			return response.Error(http.StatusBadRequest, "bad request data", err)
 		}
-		var ctx = frontendlogging.CtxVector{}
 
 		// Meta object is standard across event types, adding it globally.
-		ctx = event.AddMetaToContext(ctx)
 
 		if event.Logs != nil && len(event.Logs) > 0 {
+			var ctx = frontendlogging.CtxVector{}
+			ctx = event.AddMetaToContext(ctx)
 			for _, logEntry := range event.Logs {
-				ctx = append(ctx, "original_timestamp", logEntry.Timestamp)
-				if logEntry.LogLevel != "" {
-					ctx = append(ctx, "original_log_level", logEntry.LogLevel)
-				}
+				ctx = append(ctx, "kind", "log", "original_timestamp", logEntry.Timestamp)
+
 				for k, v := range frontendlogging.KeyValToInterfaceMap(logEntry.KeyValContext()) {
 					ctx = append(ctx, k, v)
 				}
-				grafanaJavascriptAgentLogger.Info(logEntry.Message, ctx...)
+				switch logEntry.LogLevel {
+				case frontendlogging.LogLevelDebug:
```

Comment:
`trace` level is missing; given that it can't be mapped directly, it should probably fall under `Debug` rather than `Info`

Reply:
https://github.com/grafana/grafana/pull/50801/commits/66f0908cbba8dc42fbe5d47f4425195ed2e6e199